lapis2/src/test/kotlin/org/genspectrum/lapis/model/VariantQueryFacadeTest.kt

```diff
@@ -0,0 +1,308 @@
+package org.genspectrum.lapis.model
+
+import org.genspectrum.lapis.silo.And
+import org.genspectrum.lapis.silo.Maybe
+import org.genspectrum.lapis.silo.NOf
+import org.genspectrum.lapis.silo.Not
+import org.genspectrum.lapis.silo.NucleotideSymbolEquals
+import org.genspectrum.lapis.silo.Or
+import org.genspectrum.lapis.silo.PangoLineageEquals
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class VariantQueryFacadeTest {
+    private lateinit var underTest: VariantQueryFacade
+
+    @BeforeEach
+    fun setup() {
+        underTest = VariantQueryFacade()
+    }
+
+    @Test
+    fun `given a complex variant query then map should return the corresponding SiloQuery`() {
+        val variantQuery = "300G & (400- | 500B) & !600 & MAYBE(700B | 800-) & [3-of: 123A, 234T, 345G] & A.1.2.3*"
```

Comment:
Let's maybe also add an end-to-end test that sends such a variant query to SILO that contains many filters to test the integration against SILO?

Currently, when sending this query, I get "Unknown object filter type 'NOf'" from SILO.

Reply:
I added the e2e test. However, there are two SILO bugs with Maybe and Or. Both of them are addressed by https://github.com/GenSpectrum/LAPIS-SILO/pull/93 . So we have to wait on them to make the build green again :)